### PR TITLE
Fix  #5990: Menu item alias to menu external url is not working

### LINKED ISSUE #5990: Menú item alias to menú external url is not working
#### Steps to reproduce the issue
- Create a menú item url, called "External", pointing to http://joomla.org
- Create a menú item alias, pointing to "External"

#### Expected result
- A correct redirection

#### Actual result
- A broken link to something like http://mydomain.com/?Itemid=119

#### System information (as much as possible)
Joomla 3.3.6
index.php removed from url

### PR DESCRIPTION
**Pull request by joomdonation**
Please see https://github.com/joomla/joomla-cms/issues/5990 for description of the issue.

To test it:
1. Create a menu item url, called "External", pointing to http://joomla.org for example
2. Create a menu item alias of the above menu item.
3. Apply this PR
4. Check and make sure the two menu items above work well.

### CODE REVIEW

File: modules/mod_menu/helper.php [hunk 1]
```diff
@@ -97,8 +97,16 @@ public static function getList(&$params)
 							break;
 
 						case 'alias':
-							// If this is an alias use the item id stored in the parameters to make the link.
-							$item->flink = 'index.php?Itemid=' . $item->params->get('aliasoptions');
+							$aliasItem = $menu->getItem($item->params->get('aliasoptions'));
```
Comment (zero-24) [inline modules/mod_menu/helper.php]: @joomdonation can you add one clear line bevor the if?

Comment (joomdonation) [inline modules/mod_menu/helper.php]: @zero-24 : Is it needed ? Travis is happy, so I think code style is OK with this PR ?

Comment (mbabker) [inline modules/mod_menu/helper.php]: It's supposed to be there.  Before a statement like `if`, `foreach`, or `switch`, there should be a blank line preceding it except for a comment line.  PHPCS and our ruleset isn't perfect so not all of these issues are caught.

File: modules/mod_menu/helper.php [hunk 2]
```diff
@@ -97,8 +97,16 @@ public static function getList(&$params)
...
+							if ($aliasItem->type == 'url')
+							{
+								$item->flink = $aliasItem->link;
+							}
+							else
+							{
+								// If this is an alias use the item id stored in the parameters to make the link.
+								$item->flink = 'index.php?Itemid=' . $item->params->get('aliasoptions');
+							}
 							break;
```
Comment (mbabker) [inline modules/mod_menu/helper.php]: Should also be a blank line after the closing bracket in a if/else.  I have a feeling we're still excluding some files from some checks and in general need to update our PHPCS ruleset in this repo.

File: modules/mod_menu/helper.php (hunk 1 above)
Comment (zero-24) [inline modules/mod_menu/helper.php]: @joomdonation this is common practice here see: http://joomla.github.io/coding-standards/?coding-standards/chapters/php.md I don't think this is checked by Travis.

**Comment by joomdonation** [comment]
OK. I added blank line before if clause as requested. Thanks @zero-24  and @mbabker . Do we still need blank line  after the closing bracket in an if/else?

**Comment by patrickzemke** [comment]
2 Menupoints and 2 alias create
works fine<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Comment by emstrasser** [comment]
@jtester 
with joomla 3.4.1 dev 
2 external url enu items and 2 menu item aliases testet
works fine<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Comment by arrocgn** [comment]
with Joomla! 3.4.1-dev Development
2 external + 2 menu items alias created and testet.
error without patch was seen
with new patch the links works wonderful
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Comment by msdevcon** [comment]
@Test
Patch works fine! Thanks!

**Comment by stefeschefe** [comment]
Works fine with Joomla! 3.4.1-dev Development<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Comment by brianteeman** [comment]
Multiple good tests - thanks - setting RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Comment by phproberto** [comment]
Merged and added the suggested spaces in a different commit. Thanks!

**Comment by ayito** [comment]
I have the issue, my settings are like this:
Joomla! 3.4.6 Stable [ Ember ] 15-December-2015 11:11 GMT 

1 menu item CLIENTS "Menu Item Alias" (belongs to Main Menu, access public) that points to menu item "CLIENTS" "External URL" (belongs to menu CLIENTES, not Main Menu, access registered) and the external URL don't open.

Could you please test this set up.
1.- Menu item from Main Menu with public access, type "Menu Item Alias" to
2.- Menu item from "other menu" with access registered, type "External URL" and see if it works.

Thanks!
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Comment by janinef** [comment]
I have Joomla Version 3.4.8 and alle the System menu item doesn't work. I have installed this patch but it is still not working. I always get the error message 'Error when loading the formular data' when I want to set up the menu item. Is this the same error? What can I do?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Comment by brianteeman** [comment]
This was already in 3.4.8 Please use the forum for support http://forum.joomla.org<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/5995">issues.joomla.org/joomla-cms/5995</a>.</sub>

**Pull request closed without merging**